packages/cli-kit/src/common/result.ts

```diff
@@ -0,0 +1,37 @@
+export type Result<T, TError = Error> = {ok: true; value: T} | {ok: false; error: TError}
+
+export type ResultAsync<T, TError = Error> = Promise<Result<T, TError>>
+
+export const ok = <T, TError = Error>(value: T): Result<T, TError> => {
+  return {ok: true, value}
+}
+
+const UnknownError = new Error('Unknown error')
+
+export const err = <T = never>(error?: unknown): Result<T, Error> => {
```

Comment:
Why are we making "an error that don't wrap errors" a valid state?

Reply:
To make the using of the method more flexible and easier. 
- You can pass your Error,
- You can pass a message to construct an Error
- You can use the method without parameters you a generic error is constructed